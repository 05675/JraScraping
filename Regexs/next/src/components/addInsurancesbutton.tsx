import React from 'react';
import { StyledButton } from '@src/styles';
import { StyledPlusSVG } from '@src/styles/svg';

type AddInsurancesButtonProps = {
  label?: string;
  important?: boolean;
  primary?: boolean;
  warning?: boolean;
  smallSize?: boolean;
  fontSize?: string;
  width?: string;
  height?: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onClick?: (event: any) => void;
};

export const AddInsurancesButton: React.FC<AddInsurancesButtonProps> = props => {
  return (
    <>
      <div
        style={{
          position: 'relative',
          width: `${props.width ?? '100%'}`,
          height: `${props.height ?? '48px'}`,
        }}>
        <span className='plus-icon'>
          <StyledPlusSVG color='#525252' />
        </span>
        <StyledButton
          important={props.important}
          primary={props.primary}
          warning={props.warning}
          smallSize={props.smallSize}
          fontSize={props.fontSize}
          width={props.width}
          height={props.height}
          onClick={props.onClick}
          labelOffsetLeft='48px'
          noShadow>
          {props.children}
        </StyledButton>
      </div>

      <style jsx>
        {`
          .plus-icon {
            position: absolute;
            width: 24px;
            height: 24px;
            top: calc(50% - 12px);
            left: 24px;
          }
        `}
      </style>
    </>
  );
};

export default AddInsurancesButton;
