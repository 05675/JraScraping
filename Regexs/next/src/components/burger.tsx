import React from 'react';
import { StyledBurger } from '@src/styles/burger';

/**
 * バーガーボタン用インターフェース
 *
 * @interface BurgerProps
 */
interface BurgerProps {
  open: boolean;
  setOpen: (newValue: boolean) => void;
}

/**
 * バーガーボタンのヘッダーコンポーネント
 *
 * @param {BurgerProps} props {
 *     open
 *     setOpen
 * }
 * @returns {*} Reactコンポーネント
 */
export const BurgerComponent: React.FC<BurgerProps> = props => {
  return (
    <StyledBurger open={props.open} onClick={() => props.setOpen(!props.open)}>
      <div />
      <div />
      <div />
    </StyledBurger>
  );
};
