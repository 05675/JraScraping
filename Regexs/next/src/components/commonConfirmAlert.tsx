import React from 'react';
import { confirmAlert } from 'react-confirm-alert';

interface AlertButton {
  label: string;
  labelColor?: string;
  clickHandler?: Function;
}

interface CommonConfirmAlertProps {
  message: string;
  leftButton: AlertButton;
  rightButton: AlertButton;
}
export const CommonConfirmAlert = ({
  message,
  leftButton,
  rightButton,
}: CommonConfirmAlertProps) => () =>
  confirmAlert({
    customUI: ({ onClose }) => {
      return (
        <div className='react-confirm-alert-body'>
          <p>{message}</p>
          <div className='react-confirm-alert-button-group'>
            <button
              style={{ color: leftButton.labelColor || '#525252' }}
              onClick={() => (leftButton.clickHandler && leftButton.clickHandler()) || onClose()}>
              {leftButton.label}
            </button>
            <button
              style={{ color: rightButton.labelColor || '#525252' }}
              onClick={() => (rightButton.clickHandler && rightButton.clickHandler()) || onClose()}>
              {rightButton.label}
            </button>
          </div>
        </div>
      );
    },
  });
