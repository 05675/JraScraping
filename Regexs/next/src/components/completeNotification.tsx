import React from 'react';

type notificationProps = {
  message: string;
  isNotification: boolean;
  setIsNotification: (newValue: boolean) => void;
};

export const CompleteNotification: React.FC<notificationProps> = (props) => {
  if (props.isNotification) setTimeout(() => props.setIsNotification(false), 4000);
  const handleClick = () => props.setIsNotification(!props.isNotification);
  return (
    <>
      <div className='notif-container' onClick={handleClick} role='button'>
        <p>{props.message}</p>
        <style jsx>
          {`
            .notif-container {
              width: 100%;
              height: 48px;
              display: flex;
              border-radius: 8px;
              align-items: center;
              background: #525252;
              text-align-center;
              transform: ${props.isNotification ? 'translateY(0)' : 'translateY(100%)'};
              opacity: ${props.isNotification ? '1' : '0'};
              pointer-events: ${props.isNotification ? 'auto' : 'none'};
              transition: transform 0.2s ease-in-out, opacity 0.23s ease-in-out;
            }
            .notif-container p {
              font-style: normal;
              font-weight: normal;
              font-size: 12px;
              line-height: 100%;
              letter-spacing: 0.06em;
              color: #ffffff;
              padding-left: 8px;
              width: 100%;
            }
          `}
        </style>
      </div>
    </>
  );
};

export default CompleteNotification;
