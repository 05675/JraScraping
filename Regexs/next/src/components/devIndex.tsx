import * as React from 'react';
import Link from 'next/link';

// テスト用 sample
const pathnameToPageName: { [pathname: string]: string } = {
  // Developers
  '/Developers/createTask': 'タスク作成',
  '/Developers/demoTask': 'タスク一括登録',
  '/Developers/demoControl': 'デモ用管理機能',
  '/Developers/taskManage': '年末調整',

  // 従業員page
  '/signin': 'サインイン',
  '/tasks': 'タスク一覧',
  '/nencho/test_id_0xd1': '2020年分年末調整',
  '/nencho/test_id_0xd1/insurances': '保険料控除',
  '/nencho/test_id_0xd1/lifeInsuranceInputs/new': '生命保険入力',
};

const DevIndex: React.FC = () => {
  return (
    <div>
      <ul className='list-style'>
        <p style={{ textAlign: 'center', color: '#525252' }}>Sample Page（開発用）</p>
        {Object.keys(pathnameToPageName).map(pathname => (
          <div key={pathname}>
            <li className='listItem-style'>
              <Link href={pathname}>
                <a className='Anchor-style'>{pathnameToPageName[pathname]}</a>
              </Link>
            </li>
          </div>
        ))}
      </ul>
      <style jsx>
        {`
          .list-style {
            margin: 0;
            padding: 0;
            position: relative;
          }
          .listItem-style {
            color: black;
            border-left: solid 8px orange;
            background: whitesmoke;
            margin-bottom: 5px;
            line-height: 1.5;
            border-radius: 0 15px 15px 0;
            padding: 0.5em;
            list-style-type: none !important;
          }
          .anchor-style {
            display: block;
            padding: 5px;
          }
        `}
      </style>
    </div>
  );
};

export default DevIndex;
