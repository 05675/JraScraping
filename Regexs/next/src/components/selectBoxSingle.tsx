import React from 'react';
import Select, { components, Styles } from 'react-select';
import { StyledArrowDownSVG } from '@src/styles/svg';

/**
 * コンボボックスのデザイン指定
 */
const customStyles: Styles = {
  control: (styles, state) => ({
    ...styles,
    width: '100%',
    height: '48px',
    color: '#525252',
    background: 'rgba(1,1,1,0)',
    borderRadius: '0px',
    fontSize: '16px',
    fontStyle: 'normal',
    lineHeight: '180%',
    letterSpacing: '0.06em',
    border: 'none',
    borderBottom: '1px solid',
    borderBottomColor: (() => {
      if (state.selectProps.isError) {
        return '#fa3939';
      }
      if (state.isFocused) {
        return '#3aa0fa';
      }
      return '#ebebeb';
    })(),
    boxShadow: (() => {
      if (state.selectProps.isError && state.isFocused) {
        return '0px 1px 0px #fa3939';
      }
      if (state.isFocused) {
        return '0px 1px 0px #3aa0fa';
      }
      return 'none';
    })(),
  }),

  option: (styles, state) => ({
    ...styles,
    color: '#525252',
    backgroundColor: state.isSelected ? '#ffffff' : '#ffffff',
  }),

  indicatorSeparator: () => ({}),

  valueContainer: styles => ({
    ...styles,
    padding: '0',
  }),

  singleValue: styles => ({
    ...styles,
    margin: '0',
  }),
};

/**
 * ドロップダウンのコンポーネント指定
 *
 * @param {any} props ドロップダウンインジケータの情報
 * @returns {*} ドロップダウンコンポーネント
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const DropdownIndicator: React.FC<any> = props => {
  return (
    <>
      <div style={{ marginRight: props.selectProps.isError ? '40px' : '0' }}>
        <components.DropdownIndicator {...props}>
          <StyledArrowDownSVG color='#949494' width='12px' height='12px' />
        </components.DropdownIndicator>
      </div>
    </>
  );
};

type OptionType = {
  key: string;
  label: string;
  value: string | boolean;
};
/**
 * react-selectの単一選択ドロップダウンメニュー用インターフェース
 *
 * @interface SelectBoxSingleProps
 */
export interface SelectBoxSingleProps {
  id?: string;
  placeholder?: string;
  options: OptionType[];
  value: OptionType['value'];
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onChange: (value: any) => void;
  isError?: boolean;
}

/**
 * react-selectを利用した単一選択ドロップダウンメニューの共通コンポーネント
 *
 * @param {SelectBoxSingleProps} props {
 *     placeholder
 *     value
 *     options
 *     onChange
 * }
 * @returns {*} Reactコンポーネント
 */
export const SelectBoxSingle: React.FC<SelectBoxSingleProps> = ({
  placeholder,
  options,
  value,
  ...props
}) => {
  return (
    <Select
      isSearchable={false}
      styles={customStyles}
      components={{ DropdownIndicator }}
      placeholder={placeholder || '選択してください'}
      value={options?.filter(option => option.value === value)}
      options={options}
      isError={props.isError}
      {...props}
    />
  );
};
