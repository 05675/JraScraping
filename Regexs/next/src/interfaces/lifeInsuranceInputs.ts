import { InsuranceCategories } from '@src/model/entity';
import { ErrorInfo } from '@src/interfaces/errorInfo';
import { InputProps } from '@src/styles';

export interface NenchoInsuranceLifeInput2020 {
  id?: number;
  nenchoId: string;
  category: InsuranceCategories;
  firmName: string;
  categoryDetail: string;
  period: string;
  contractorName: string;
  receiverName: string;
  relation: string;
  payment: string;
}

type ValidateFormProps = {
  isError: boolean;
  errorMessage: string;
};

export interface NenchoInsuranceLifeInput2020Validate {
  category: ValidateFormProps;
  firmName: ValidateFormProps;
  categoryDetail: ValidateFormProps;
  period: ValidateFormProps;
  contractorName: ValidateFormProps;
  receiverName: ValidateFormProps;
  relation: ValidateFormProps;
  payment: ValidateFormProps;
}

export interface OptionType {
  key: string;
  label: string;
  value: string | boolean;
}

export interface InputWithLabelProps extends InputProps {
  label: string;
  name: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  errorMessage?: string;
}

export interface SelectBoxProps {
  label: string;
  options: OptionType[];
  value: OptionType['value'];
  onChange: (selectedOption: OptionType) => void;
}

export interface LifeInsuranceInputsProps {
  isUpdatePage: boolean;
  nenchoId: string | string[];
  lifeInsuranceData?: NenchoInsuranceLifeInput2020;
  token?: string;
  errorInfo?: ErrorInfo;
}
