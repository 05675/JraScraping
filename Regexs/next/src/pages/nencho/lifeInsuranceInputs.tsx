import React, { useState } from 'react';
import { NextPage } from 'next';
import { StyledLabel, StyledText } from '@src/styles/label';
import { SingleSelectComponent } from '@src/components/singleSelection';
import { StyledInput } from '@src/styles';
import { withAuthSync } from '@src/util/auth';
import { PageHeader } from '@src/components/pageHeader';

type OptionType = {
  label: string;
  value: string;
  groupId?: string;
};

// TODO:リストアイテム仮置き
interface ListItem {
  item: OptionType[];
}

const Form: React.FC<{ label: string }> = ({ label }) => {
  return (
    <>
      <StyledLabel>{label}</StyledLabel>
      <StyledInput style={{ margin: '6px auto 16px auto' }} />
    </>
  );
};

const ListBox: React.FC<{ label: string; items: ListItem }> = ({ label, items }) => {
  return (
    <>
      <StyledLabel> {label} </StyledLabel>
      <SingleSelectComponent options={items.item} onChange={useState} />
    </>
  );
};

/**
 * 生命保険入力画面
 *
 * @returns {*} Reactコンポーネント
 */
const LifeInsuranceInputs: NextPage = () => {
  const saveLifeInsuranceInfo = () => {
    // FIXME: 2020-02-26 K.TANAKA DBへの登録処理をここに実装する
    // eslint-disable-next-line no-alert
    alert('入力内容をDBに登録');
  };

  const insList: ListItem = {
    item: [],
  };
  // const [insuranceType, setInsuranceTypeList] = useState();
  return (
    <>
      <PageHeader title='生命保険' buttonName='保存' buttonFunc={() => saveLifeInsuranceInfo()} />

      <div style={{ margin: '0 16px' }}>
        <ListBox label='生命保険の種類' items={insList} />
        <Form label='保険会社当の名称' />
        <Form label='保険等の種類' />
        <Form label='保険期間' />
        <Form label='保険等の契約者の氏名' />
        <StyledLabel>保険金等の受取人</StyledLabel>
        <div style={{ display: 'flex' }}>
          <div style={{ width: '50%' }}>
            <Form label='氏名' />
          </div>
          <div style={{ width: '50%', marginLeft: '16px' }}>
            <Form label='あなたとの続柄' />
          </div>
        </div>
        <ListBox label='新・旧の区分' items={insList} />
        <Form label='申告額' />
        <StyledText>
          あなたが本年中に支払った保険料等の金額（分配を受けた剰余金等の控除後の金額）
        </StyledText>
      </div>
    </>
  );
};

export default withAuthSync(LifeInsuranceInputs);
