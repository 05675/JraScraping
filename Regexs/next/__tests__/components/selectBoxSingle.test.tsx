import React from 'react';
import renderer from 'react-test-renderer';
import { SelectBoxSingle } from '@src/components/selectBoxSingle';
import { OptionType } from '@src/interfaces/lifeInsuranceInputs';

describe('SelectBoxSingle', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const handleChange: (value: any) => void = jest.fn();
  const option: OptionType[] = [];

  test('snapshot isError', () => {
    const snapshot = renderer
      .create(
        <SelectBoxSingle
          placeholder='label'
          options={option}
          value='value'
          onChange={handleChange}
          isError
        />
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('snapshot isNotError', () => {
    const snapshot = renderer
      .create(
        <SelectBoxSingle
          placeholder='label'
          options={option}
          value='value'
          onChange={handleChange}
          isError={false}
        />
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
