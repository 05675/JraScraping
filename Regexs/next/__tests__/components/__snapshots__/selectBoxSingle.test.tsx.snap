// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBoxSingle snapshot isError 1`] = `
<div
  className=" css-2b097c-container"
  onKeyDown={[Function]}
>
  <div
    className=" css-y6gifx-control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className=" css-86rg1c-ValueContainer"
    >
      <div
        className=" css-1wa3eu0-placeholder"
      >
        label
      </div>
      <input
        aria-autocomplete="list"
        className="css-62g3xt-dummyInput"
        disabled={false}
        id="react-select-2-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        tabIndex="0"
        value=""
      />
    </div>
    <div
      className=" css-1hb7zxy-IndicatorsContainer"
    >
      <span
        className=" css-62wm9a-IndicatorSeparator"
      />
      <div
        style={
          Object {
            "marginRight": "40px",
          }
        }
      >
        <div
          aria-hidden="true"
          className=" css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg>

            className="sc-AxjAm fwkFKF"
            color="#949494"
            height="12px"
            width="12px"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectBoxSingle snapshot isNotError 1`] = `
<div
  className=" css-2b097c-container"
  onKeyDown={[Function]}
>
  <div
    className=" css-88o9p8-control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className=" css-86rg1c-ValueContainer"
    >
      <div
        className=" css-1wa3eu0-placeholder"
      >
        label
      </div>
      <input
        aria-autocomplete="list"
        className="css-62g3xt-dummyInput"
        disabled={false}
        id="react-select-3-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        tabIndex="0"
        value=""
      />
    </div>
    <div
      className=" css-1hb7zxy-IndicatorsContainer"
    >
      <span
        className=" css-62wm9a-IndicatorSeparator"
      />
      <div
        style={
          Object {
            "marginRight": "0",
          }
        }
      >
        <div
          aria-hidden="true"
          className=" css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg>

            className="sc-AxjAm fwkFKF"
            color="#949494"
            height="12px"
            width="12px"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
